sys/include/net/gnrc/ipv6/nib.h

```diff
@@ -178,8 +184,109 @@ extern "C" {
+ * @return  -ENETUNREACH if there is no route to host.
+ * @return  -EHOSTUNREACH if the next hop is not reachable or if @p dst was
+ *          link-local, but @p iface was @ref KERNEL_PID_UNDEF (no neighbor
+ *          cache entry will be created in this case and no neighbor
+ *          solicitation sent).
+ */
+int gnrc_ipv6_nib_get_next_hop_l2addr(const ipv6_addr_t *dst,
+                                      kernel_pid_t iface, gnrc_pktsnip_t *pkt,
+                                      gnrc_ipv6_nib_nc_t *nce);
+
+/**
+ * @brief   Handle a received ICMPv6 packet
+ *
+ * @pre `iface != KERNEL_PID_UNDEF`
+ * @pre `ipv6 != NULL`
+ * @pre `icmpv6 != NULL`
+ * @pre `icmpv6_len > sizeof(icmpv6_hdr_t)`
+ *
+ * @attention   The ICMPv6 checksum is supposed to be checked externally!
+ *
+ * @note    @p ipv6 is just used for the addresses and hop limit. The next
+ *          header field will not be checked for correctness (but should be
+ *          @ref PROTNUM_ICMPV6)
+ *
+ * @see [RFC 4861, section 6.1](https://tools.ietf.org/html/rfc4861#section-6.1)
+ * @see [RFC 4861, section 6.2.6](https://tools.ietf.org/html/rfc4861#section-6.2.6)
+ * @see [RFC 4861, section 6.3.4](https://tools.ietf.org/html/rfc4861#section-6.3.4)
+ * @see [RFC 4861, section 7.1](https://tools.ietf.org/html/rfc4861#section-7.1)
+ * @see [RFC 4861, section 7.2.3](https://tools.ietf.org/html/rfc4861#section-7.2.3)
+ * @see [RFC 4861, section 7.2.5](https://tools.ietf.org/html/rfc4861#section-7.2.5)
+ * @see [RFC 4861, section 8.1](https://tools.ietf.org/html/rfc4861#section-8.1)
+ * @see [RFC 4861, section 8.3](https://tools.ietf.org/html/rfc4861#section-8.3)
+ * @see [RFC 4862, section 5.4.3](https://tools.ietf.org/html/rfc4862#section-5.4.3)
+ * @see [RFC 4862, section 5.4.4](https://tools.ietf.org/html/rfc4862#section-5.4.4)
+ * @see [RFC 4862, section 5.5.3](https://tools.ietf.org/html/rfc4862#section-5.5.3)
+ * @see [RFC 6775, section 5.5.2](https://tools.ietf.org/html/rfc6775#section-5.5.2)
+ * @see [RFC 6775, section 5.4](https://tools.ietf.org/html/rfc6775#section-5.4)
+ * @see [RFC 6775, section 6.3](https://tools.ietf.org/html/rfc6775#section-6.3)
+ * @see [RFC 6775, section 6.5](https://tools.ietf.org/html/rfc6775#section-6.5)
+ * @see [RFC 6775, section 8.1.3](https://tools.ietf.org/html/rfc6775#section-8.1.3)
+ * @see [RFC 6775, section 8.2.1](https://tools.ietf.org/html/rfc6775#section-8.2.1)
+ * @see [RFC 6775, section 8.2.4](https://tools.ietf.org/html/rfc6775#section-8.2.4)
+ * @see [RFC 6775, section 8.2.5](https://tools.ietf.org/html/rfc6775#section-8.2.5)
+ *
+ * @param[in] iface         The interface the packet came over.
+ * @param[in] ipv6          The IPv6 header of the received packet.
+ * @param[in] icmpv6        The ICMPv6 header and payload of the received
+ *                          packet.
+ * @param[in] icmpv6_len    The number of bytes at @p icmpv6.
+ */
+void gnrc_ipv6_nib_handle_pkt(kernel_pid_t iface, const ipv6_hdr_t *ipv6,
```

Comment:
the function only handles icmpv6. should we put that info somewhere into the name of the func?

Reply:
I think the function name is sufficiently long. If someone does not check by here that the *neighbor* information base only concerns NDP (and thus ICMPv6) they should not meddle with this function anyway.